### ISSUE TITLE
Error when change subscriber requests list of changes after receiving commit notification

### ISSUE THREAD

**Issue by alangefe** [sysrepo/sysrepo#1113]
### Description
When a subscriber who has been sent a commit notification requests the list of changes, sometimes the subscriber will get an error indicating that the commit context does not exist.

The problem is that sysrepo sends out the commit notifications prior to saving the commit context, meaning there is a window of time during which the subscriber can ask sysrepo for the changes before they have been saved.

The diagram below depicts the erroneous behavior:

```
    Errorneous case:

+----------+                  +-----------------+
| sysrepod |                  |change subscriber|
+-----+----+                  +--------+--------+
      |                                |
      |       Notify about changes     |
      +-------------------------------->
      |                                |
      |   Request list of changes      |
      <--------------------------------+
      |                                |
      |    Error: No commit context    |
      +-------------------------------->
      |                                |
      |                                |
      +--+                             |
      |  | Save commit context         |
      |  |                             |
      <--+                             |
      |                                |
      v                                v
```

### Proposed solution
We've implemented a proposed solution in https://github.com/alangefe/sysrepo/commit/06cd6c0d4f66c40a1a354278cb872bbea35684aa. This commit postpones sending the commit notifications until after the commit context has been saved. In other words:

```
+----------+                  +-----------------+
| sysrepod |                  |change subscriber|
+----+-----+                  +-------+---------+
     |                                |
     +--+                             |
     |  | Save commit context         |
     |  |                             |
     <--+                             |
     |                                |
     |                                |
     |       Notify about changes     |
     +-------------------------------->
     |                                |
     |   Request list of changes      |
     <--------------------------------+
     |                                |
     |   Return changes from context  |
     +-------------------------------->
     |                                |
     v                                v
```

**Comment by michalvasko** [comment]
Proposed solution merged, should not happen any more.

Regards,
Michal

**Comment by alangefe** [comment]
Closed with #1114, included in 0.7.4.
